### PR TITLE
Add new ElasticSearch check

### PR DESCRIPTION
**Pull request by bram123**
Signed-off-by: Bram Leeda <bram@123inkt.nl>

|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | no

### Description

Check for cluster connection
Check for cluster health

Returns responsetime along with success/warning/failure.

### CODE REVIEW

**Comment by bram123** [comment]
The new check supports both elasticsearch-php 7 and 8.
- Version 7 is what I personally need, but I added 8 as well for completeness.

Problem is that they changed namespace, the check and test support both and I tested locally with both.
But psalm only knows about the version that was installed, not the other.
- Not too sure on how to fix this, add stub files for both versions?

File: composer.json [hunk 1]
```diff
@@ -38,14 +38,15 @@
     },
     "require-dev": {
         "doctrine/migrations": "^2.0 || ^3.0",
+        "elasticsearch/elasticsearch": "^7.17 || ^8.5",
```
Comment (Ocramius) [inline composer.json]: I don't think we should use the `elasticsearch/elasticsearch` package in order to query its status: elasticsearch can/should communicate with its REST-ish API

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (bram123) [reply composer.json]: That makes my life/PR a lot easier

File: test/ElasticSearchTest.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+     * @param class-string<ResultInterface> $expectedResult
+     * @throws Exception
+     */
+    public function testElasticSearch(string $clusterStatus, string $expectedResult): void
+    {
+        $mockHandler = new MockHandler([new Response(200, [], $clusterStatus)]);
+        $mockClient  = new Client(['handler' => $mockHandler]);
+        $check       = new ElasticSearch('localhost:9200', [], [], $mockClient);
+
+        static::assertInstanceOf($expectedResult, $check->check());
```
Comment (Ocramius) [inline test/ElasticSearchTest.php]: One last thing: let's verify that the `responseTime` is provided and is a number :)

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Review by Ocramius — APPROVED** [review]
Excellent, thanks @bram123!

**Comment by Ocramius** [comment]
Released BTW :+1:

**Comment by bram123** [comment]
@Ocramius It looks like this PR was merged into 1.20.x correctly, but it's not present in 1.21.x and later branches/releases.
Should I create a new PR with the same changes?

**Comment by Ocramius** [comment]
Argh, the target branch was wrong.

No, in that case please make a new PR (same commits OK) against the latest branch, and I'll re-release.

We'll then have to rollback `1.20.x`

**Comment by bram123** [comment]
Submitted new PR: https://github.com/laminas/laminas-diagnostics/pull/66

**Pull request merged**